python/xorbits/core/adapter.py

```diff
@@ -139,6 +142,149 @@ def decorate(cls: Type):
     return decorate
 
 
+class ClsMethodWrapper(ABC):
+    def __init__(
+        self,
+        func_name: str = "",
+        library_cls: Type = object,
+        fallback_warning: bool = False,
+    ):
+        self.library_cls = library_cls
+        self.func_name = func_name
+        self.fallback_warning = fallback_warning
+
+    @abstractmethod
+    def generate_fallback_data(self, mars_entity: MarsEntity):
+        """
+        let mars entity fallback to data according to the library
+
+        Parameters
+        ----------
+        mars_entity
+
+        Returns
+        -------
+
+        """
+
+    @abstractmethod
+    def generate_warning_msg(self, mars_entity: MarsEntity, func_name: str):
```

Comment:
```suggestion
    def generate_warning_msg(self, mars_entity: MarsEntity, func_name: str) -> str:
```

Don't forget to add return type as well, so do others.